src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -51,81 +52,50 @@
 
 
     /**
      * Constructor.
      *
-     * @param topics                      Root Configuration topic for this service
-     * @param groupManager                Group configuration management
-     * @param certificateManager          Certificate management
-     * @param authorizationHandler        authorization handler for IPC calls
-     * @param greengrassCoreIPCService    core IPC service
-     * @param mqttSessionFactory          session factory to handling mqtt credentials
-     * @param sessionManager              session manager
-     * @param clientDevicesAuthServiceApi client devices service api handle
-     * @param useCases                    UseCases service
+     * @param topics Root Configuration topic for this service
      */
-    @SuppressWarnings("PMD.ExcessiveParameterList")
     @Inject
-    public ClientDevicesAuthService(Topics topics, GroupManager groupManager,
-                                    CertificateManager certificateManager,
-                                    AuthorizationHandler authorizationHandler,
-                                    GreengrassCoreIPCService greengrassCoreIPCService,
-                                    MqttSessionFactory mqttSessionFactory,
-                                    SessionManager sessionManager,
-                                    ClientDevicesAuthServiceApi clientDevicesAuthServiceApi,
-                                    UseCases useCases) {
+    public ClientDevicesAuthService(Topics topics) {
         super(topics);
-        cloudCallQueueSize = DEFAULT_CLOUD_CALL_QUEUE_SIZE;
-        cloudCallQueueSize = getValidCloudCallQueueSize(topics);
-        cloudCallThreadPool = new ThreadPoolExecutor(1,
-                DEFAULT_THREAD_POOL_SIZE, 60, TimeUnit.SECONDS,
-                new ResizableLinkedBlockingQueue<>(cloudCallQueueSize));
-        cloudCallThreadPool.allowCoreThreadTimeOut(true); // act as a cached threadpool
-        this.clientDevicesAuthServiceApi = clientDevicesAuthServiceApi;
-        this.groupManager = groupManager;
-        this.certificateManager = certificateManager;
-        this.authorizationHandler = authorizationHandler;
-        this.greengrassCoreIPCService = greengrassCoreIPCService;
-        SessionCreator.registerSessionFactory("mqtt", mqttSessionFactory);
-        certificateManager.updateCertificatesConfiguration(new CertificatesConfig(getConfig()));
-        sessionManager.setSessionConfig(new SessionConfig(getConfig()));
-
-        // Initialize the use cases, so we can use them anywhere in the app
-        this.useCases = useCases;
-        this.useCases.init(getContext());
+        this.context = topics.getContext();
     }
 
+    @Override
+    protected void install() throws InterruptedException {
+        super.install();
+
+        context.get(UseCases.class).init(context);
+        context.get(CertificateManager.class).updateCertificatesConfiguration(new CertificatesConfig(getConfig()));
```

Comment:
getting stuff from the context may fail with a runtime exception if the class (or a class that is needed by that class) cannot be constructed. Maybe this can't be an issue, but just FYI

Reply:
Yes, though the results will be a bit different. If it failed in your constructor, then your service would be replaced with `UnloadableService`. Whereas here, it will fail while starting which will end with the service in the broken state.